Build fails with node 0.11 on MacBook Pro Maverick

Build is failing with node 0.11 on my MacBook Pro.  It works on node version 0.10.  Below is the output from the console:

```
$ npm install -g serialport
npm http GET https://registry.npmjs.org/serialport
npm http 304 https://registry.npmjs.org/serialport
npm http GET https://registry.npmjs.org/node-pre-gyp
npm http GET https://registry.npmjs.org/bindings/1.1.1
npm http GET https://registry.npmjs.org/async/0.1.18
npm http GET https://registry.npmjs.org/sf/0.1.6
npm http GET https://registry.npmjs.org/optimist
npm http GET https://registry.npmjs.org/nan
npm http 304 https://registry.npmjs.org/optimist
npm http 304 https://registry.npmjs.org/nan
npm http 304 https://registry.npmjs.org/bindings/1.1.1
npm http 304 https://registry.npmjs.org/node-pre-gyp
npm http 304 https://registry.npmjs.org/sf/0.1.6
npm http 304 https://registry.npmjs.org/async/0.1.18
npm http GET https://registry.npmjs.org/wordwrap
npm http 304 https://registry.npmjs.org/wordwrap
/usr/local/share/npm/bin/serialportlist -> /usr/local/share/npm/lib/node_modules/serialport/bin/serialportList.js
/usr/local/share/npm/bin/serialportterm -> /usr/local/share/npm/lib/node_modules/serialport/bin/serialportTerminal.js

> serialport@1.4.2 install /usr/local/share/npm/lib/node_modules/serialport
> node-pre-gyp install --fallback-to-build

node-pre-gyp http GET https://node-serialport.s3.amazonaws.com/serialport/v1.4.2/Release/node-v14-darwin-x64.tar.gz
node-pre-gyp http 404 https://node-serialport.s3.amazonaws.com/serialport/v1.4.2/Release/node-v14-darwin-x64.tar.gz
node-pre-gyp http Pre-built binary not available for your system (falling back to source compile with node-gyp) 
  CXX(target) Release/obj.target/serialport/src/serialport.o
In file included from ../src/serialport.cpp:3:
In file included from ../src/serialport.h:5:
../node_modules/nan/nan.h:360:13: error: no member named 'New' in 'v8::String'
    return  _NAN_ERROR(v8::Exception::Error, errmsg);
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:340:50: note: expanded from macro '_NAN_ERROR'
# define _NAN_ERROR(fun, errmsg) fun(v8::String::New(errmsg))
                                     ~~~~~~~~~~~~^
../node_modules/nan/nan.h:364:5: error: no member named 'ThrowException' in
      namespace 'v8'
    _NAN_THROW_ERROR(v8::Exception::Error, errmsg);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:345:11: note: expanded from macro '_NAN_THROW_ERROR'
      v8::ThrowException(_NAN_ERROR(fun, errmsg));                             \
      ~~~~^
../node_modules/nan/nan.h:364:5: error: no member named 'New' in 'v8::String'
    _NAN_THROW_ERROR(v8::Exception::Error, errmsg);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:345:26: note: expanded from macro '_NAN_THROW_ERROR'
      v8::ThrowException(_NAN_ERROR(fun, errmsg));                             \
                         ^~~~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:340:50: note: expanded from macro '_NAN_ERROR'
# define _NAN_ERROR(fun, errmsg) fun(v8::String::New(errmsg))
                                     ~~~~~~~~~~~~^
../node_modules/nan/nan.h:369:9: error: no type named 'ThrowException' in
      namespace 'v8'
    v8::ThrowException(error);
    ~~~~^
../node_modules/nan/nan.h:376:65: error: no member named 'New' in 'v8::String'
    v8::Local<v8::Value> err = v8::Exception::Error(v8::String::New(msg));
                                                    ~~~~~~~~~~~~^
../node_modules/nan/nan.h:378:26: error: no member named 'New' in 'v8::String'
    obj->Set(v8::String::New("code"), v8::Int32::New(errorNumber));
             ~~~~~~~~~~~~^
../node_modules/nan/nan.h:390:12: error: no member named 'New' in 'v8::String'
    return _NAN_ERROR(v8::Exception::TypeError, errmsg);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:340:50: note: expanded from macro '_NAN_ERROR'
# define _NAN_ERROR(fun, errmsg) fun(v8::String::New(errmsg))
                                     ~~~~~~~~~~~~^
../node_modules/nan/nan.h:394:5: error: no member named 'ThrowException' in
      namespace 'v8'
    _NAN_THROW_ERROR(v8::Exception::TypeError, errmsg);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:345:11: note: expanded from macro '_NAN_THROW_ERROR'
      v8::ThrowException(_NAN_ERROR(fun, errmsg));                             \
      ~~~~^
../node_modules/nan/nan.h:394:5: error: no member named 'New' in 'v8::String'
    _NAN_THROW_ERROR(v8::Exception::TypeError, errmsg);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:345:26: note: expanded from macro '_NAN_THROW_ERROR'
      v8::ThrowException(_NAN_ERROR(fun, errmsg));                             \
                         ^~~~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:340:50: note: expanded from macro '_NAN_ERROR'
# define _NAN_ERROR(fun, errmsg) fun(v8::String::New(errmsg))
                                     ~~~~~~~~~~~~^
../node_modules/nan/nan.h:398:12: error: no member named 'New' in 'v8::String'
    return _NAN_ERROR(v8::Exception::RangeError, errmsg);
           ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:340:50: note: expanded from macro '_NAN_ERROR'
# define _NAN_ERROR(fun, errmsg) fun(v8::String::New(errmsg))
                                     ~~~~~~~~~~~~^
../node_modules/nan/nan.h:402:5: error: no member named 'ThrowException' in
      namespace 'v8'
    _NAN_THROW_ERROR(v8::Exception::RangeError, errmsg);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:345:11: note: expanded from macro '_NAN_THROW_ERROR'
      v8::ThrowException(_NAN_ERROR(fun, errmsg));                             \
      ~~~~^
../node_modules/nan/nan.h:402:5: error: no member named 'New' in 'v8::String'
    _NAN_THROW_ERROR(v8::Exception::RangeError, errmsg);
    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:345:26: note: expanded from macro '_NAN_THROW_ERROR'
      v8::ThrowException(_NAN_ERROR(fun, errmsg));                             \
                         ^~~~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:340:50: note: expanded from macro '_NAN_ERROR'
# define _NAN_ERROR(fun, errmsg) fun(v8::String::New(errmsg))
                                     ~~~~~~~~~~~~^
../node_modules/nan/nan.h:712:49: error: too few arguments to function call,
      single argument 'isolate' was not specified
    v8::Local<v8::Object> obj = v8::Object::New();
                                ~~~~~~~~~~~~~~~ ^
/Users/bgoode/.node-gyp/0.11.13/deps/v8/include/v8.h:2457:3: note: 'New'
      declared here
  static Local<Object> New(Isolate* isolate);
  ^
In file included from ../src/serialport.cpp:3:
In file included from ../src/serialport.h:5:
../node_modules/nan/nan.h:718:49: error: too few arguments to function call,
      single argument 'isolate' was not specified
    v8::Local<v8::Object> obj = v8::Object::New();
                                ~~~~~~~~~~~~~~~ ^
/Users/bgoode/.node-gyp/0.11.13/deps/v8/include/v8.h:2457:3: note: 'New'
      declared here
  static Local<Object> New(Isolate* isolate);
  ^
In file included from ../src/serialport.cpp:3:
In file included from ../src/serialport.h:5:
../node_modules/nan/nan.h:725:12: error: no member named 'Dispose' in
      'v8::Persistent<v8::Object, v8::NonCopyablePersistentTraits<v8::Object> >'
    handle.Dispose();
    ~~~~~~ ^
../node_modules/nan/nan.h:726:12: error: no member named 'Clear' in
      'v8::Persistent<v8::Object, v8::NonCopyablePersistentTraits<v8::Object> >'
    handle.Clear();
    ~~~~~~ ^
../node_modules/nan/nan.h:731:39: error: no member named 'NewSymbol' in
      'v8::String'; did you mean 'IsSymbol'?
    NanPersistentToLocal(handle)->Set(NanSymbol("callback"), fn);
                                      ^~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:177:38: note: expanded from macro 'NanSymbol'
#define NanSymbol(value) v8::String::NewSymbol(value)
                         ~~~~~~~~~~~~^
/Users/bgoode/.node-gyp/0.11.13/deps/v8/include/v8.h:1441:8: note: 'IsSymbol'
      declared here
  bool IsSymbol() const;
       ^
In file included from ../src/serialport.cpp:3:
In file included from ../src/serialport.h:5:
../node_modules/nan/nan.h:731:39: error: call to non-static member function
      without an object argument
    NanPersistentToLocal(handle)->Set(NanSymbol("callback"), fn);
                                      ^~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:177:38: note: expanded from macro 'NanSymbol'
#define NanSymbol(value) v8::String::NewSymbol(value)
                         ~~~~~~~~~~~~^~~~~~~~~
../node_modules/nan/nan.h:735:46: error: no member named 'NewSymbol' in
      'v8::String'; did you mean 'IsSymbol'?
    return NanPersistentToLocal(handle)->Get(NanSymbol("callback"))
                                             ^~~~~~~~~~~~~~~~~~~~~
../node_modules/nan/nan.h:177:38: note: expanded from macro 'NanSymbol'
#define NanSymbol(value) v8::String::NewSymbol(value)
                         ~~~~~~~~~~~~^
/Users/bgoode/.node-gyp/0.11.13/deps/v8/include/v8.h:1441:8: note: 'IsSymbol'
      declared here
  bool IsSymbol() const;
       ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make: *** [Release/obj.target/serialport/src/serialport.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:267:23)
gyp ERR! stack     at ChildProcess.EventEmitter.emit (events.js:110:17)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1046:12)
gyp ERR! System Darwin 13.3.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild" "--name=serialport" "--configuration=Release" "--module_name=serialport" "--version=1.4.2" "--major=1" "--minor=4" "--patch=2" "--runtime=node" "--node_abi=node-v14" "--platform=darwin" "--target_platform=darwin" "--arch=x64" "--target_arch=x64" "--module_main=./serialport" "--host=https://node-serialport.s3.amazonaws.com/" "--module_path=/usr/local/share/npm/lib/node_modules/serialport/build/serialport/v1.4.2/Release/node-v14-darwin-x64" "--remote_path=./serialport/v1.4.2/Release/" "--package_name=node-v14-darwin-x64.tar.gz" "--staged_tarball=build/stage/serialport/v1.4.2/Release/node-v14-darwin-x64.tar.gz" "--hosted_path=https://node-serialport.s3.amazonaws.com/serialport/v1.4.2/Release/" "--hosted_tarball=https://node-serialport.s3.amazonaws.com/serialport/v1.4.2/Release/node-v14-darwin-x64.tar.gz"
gyp ERR! cwd /usr/local/share/npm/lib/node_modules/serialport
gyp ERR! node -v v0.11.13
gyp ERR! node-gyp -v v0.13.0
gyp ERR! not ok 
node-pre-gyp ERR! build error 
node-pre-gyp ERR! stack Error: Failed to execute '/usr/local/bin/node rebuild --name=serialport --configuration=Release --module_name=serialport --version=1.4.2 --major=1 --minor=4 --patch=2 --runtime=node --node_abi=node-v14 --platform=darwin --target_platform=darwin --arch=x64 --target_arch=x64 --module_main=./serialport --host=https://node-serialport.s3.amazonaws.com/ --module_path=/usr/local/share/npm/lib/node_modules/serialport/build/serialport/v1.4.2/Release/node-v14-darwin-x64 --remote_path=./serialport/v1.4.2/Release/ --package_name=node-v14-darwin-x64.tar.gz --staged_tarball=build/stage/serialport/v1.4.2/Release/node-v14-darwin-x64.tar.gz --hosted_path=https://node-serialport.s3.amazonaws.com/serialport/v1.4.2/Release/ --hosted_tarball=https://node-serialport.s3.amazonaws.com/serialport/v1.4.2/Release/node-v14-darwin-x64.tar.gz' (1)
node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (/usr/local/share/npm/lib/node_modules/serialport/node_modules/node-pre-gyp/lib/util/compile.js:76:29)
node-pre-gyp ERR! stack     at ChildProcess.EventEmitter.emit (events.js:110:17)
node-pre-gyp ERR! stack     at maybeClose (child_process.js:992:16)
node-pre-gyp ERR! stack     at Process.ChildProcess._handle.onexit (child_process.js:1059:5)
node-pre-gyp ERR! System Darwin 13.3.0
node-pre-gyp ERR! command "node" "/usr/local/share/npm/lib/node_modules/serialport/node_modules/.bin/node-pre-gyp" "install" "--fallback-to-build"
node-pre-gyp ERR! cwd /usr/local/share/npm/lib/node_modules/serialport
node-pre-gyp ERR! node -v v0.11.13
node-pre-gyp ERR! node-pre-gyp -v v0.5.19
node-pre-gyp ERR! not ok 
Failed to execute '/usr/local/bin/node rebuild --name=serialport --configuration=Release --module_name=serialport --version=1.4.2 --major=1 --minor=4 --patch=2 --runtime=node --node_abi=node-v14 --platform=darwin --target_platform=darwin --arch=x64 --target_arch=x64 --module_main=./serialport --host=https://node-serialport.s3.amazonaws.com/ --module_path=/usr/local/share/npm/lib/node_modules/serialport/build/serialport/v1.4.2/Release/node-v14-darwin-x64 --remote_path=./serialport/v1.4.2/Release/ --package_name=node-v14-darwin-x64.tar.gz --staged_tarball=build/stage/serialport/v1.4.2/Release/node-v14-darwin-x64.tar.gz --hosted_path=https://node-serialport.s3.amazonaws.com/serialport/v1.4.2/Release/ --hosted_tarball=https://node-serialport.s3.amazonaws.com/serialport/v1.4.2/Release/node-v14-darwin-x64.tar.gz' (1)
npm ERR! serialport@1.4.2 install: `node-pre-gyp install --fallback-to-build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the serialport@1.4.2 install script.
npm ERR! This is most likely a problem with the serialport package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-pre-gyp install --fallback-to-build
npm ERR! You can get their info via:
npm ERR!     npm owner ls serialport
npm ERR! There is likely additional logging output above.

npm ERR! System Darwin 13.3.0
npm ERR! command "/usr/local/bin/node" "/usr/local/bin/npm" "install" "-g" "serialport"
npm ERR! cwd /Users/bgoode/automategreen/home-controller
npm ERR! node -v v0.11.13
npm ERR! npm -v 1.3.21
npm ERR! code ELIFECYCLE
npm ERR! 
npm ERR! Additional logging details can be found in:
npm ERR!     /Users/bgoode/automategreen/home-controller/npm-debug.log
npm ERR! not ok code 0
```

--- a/package.json
+++ b/package.json
@@ -47,7 +47,7 @@
     "async": "0.1.18",
     "sf": "0.1.6",
     "optimist": "~0.3.4",
-    "nan": "~0.7.0"
+    "nan": "~1.1.0"
   },
   "bundledDependencies":["node-pre-gyp"],
   "devDependencies": {

--- a/src/serialport.cpp
+++ b/src/serialport.cpp
@@ -4,7 +4,7 @@
 #include "queue.h"
 
 #ifdef WIN32
-#define strcasecmp stricmp
+#define strncasecmp strnicmp
 #else
 #include "serialport_poller.h"
 #endif
@@ -42,20 +42,20 @@ NAN_METHOD(Open) {
   OpenBaton* baton = new OpenBaton();
   memset(baton, 0, sizeof(OpenBaton));
   strcpy(baton->path, *path);
-  baton->baudRate = options->Get(v8::String::New("baudRate"))->ToInt32()->Int32Value();
-  baton->dataBits = options->Get(v8::String::New("dataBits"))->ToInt32()->Int32Value();
-  baton->bufferSize = options->Get(v8::String::New("bufferSize"))->ToInt32()->Int32Value();
-  baton->parity = ToParityEnum(options->Get(v8::String::New("parity"))->ToString());
-  baton->stopBits = ToStopBitEnum(options->Get(v8::String::New("stopBits"))->ToNumber()->NumberValue());
-  baton->rtscts = options->Get(v8::String::New("rtscts"))->ToBoolean()->BooleanValue();
-  baton->xon = options->Get(v8::String::New("xon"))->ToBoolean()->BooleanValue();
-  baton->xoff = options->Get(v8::String::New("xoff"))->ToBoolean()->BooleanValue();
-  baton->xany = options->Get(v8::String::New("xany"))->ToBoolean()->BooleanValue();
+  baton->baudRate = options->Get(NanNew<v8::String>("baudRate"))->ToInt32()->Int32Value();
+  baton->dataBits = options->Get(NanNew<v8::String>("dataBits"))->ToInt32()->Int32Value();
+  baton->bufferSize = options->Get(NanNew<v8::String>("bufferSize"))->ToInt32()->Int32Value();
+  baton->parity = ToParityEnum(options->Get(NanNew<v8::String>("parity"))->ToString());
+  baton->stopBits = ToStopBitEnum(options->Get(NanNew<v8::String>("stopBits"))->ToNumber()->NumberValue());
+  baton->rtscts = options->Get(NanNew<v8::String>("rtscts"))->ToBoolean()->BooleanValue();
+  baton->xon = options->Get(NanNew<v8::String>("xon"))->ToBoolean()->BooleanValue();
+  baton->xoff = options->Get(NanNew<v8::String>("xoff"))->ToBoolean()->BooleanValue();
+  baton->xany = options->Get(NanNew<v8::String>("xany"))->ToBoolean()->BooleanValue();
 
   baton->callback = new NanCallback(callback);
-  baton->dataCallback = new NanCallback(options->Get(NanSymbol("dataCallback")).As<v8::Function>());
-  baton->disconnectedCallback = new NanCallback(options->Get(NanSymbol("disconnectedCallback")).As<v8::Function>());
-  baton->errorCallback = new NanCallback(options->Get(NanSymbol("errorCallback")).As<v8::Function>());
+  baton->dataCallback = new NanCallback(options->Get(NanNew<v8::String>("dataCallback")).As<v8::Function>());
+  baton->disconnectedCallback = new NanCallback(options->Get(NanNew<v8::String>("disconnectedCallback")).As<v8::Function>());
+  baton->errorCallback = new NanCallback(options->Get(NanNew<v8::String>("errorCallback")).As<v8::Function>());
 
   uv_work_t* req = new uv_work_t();
   req->data = baton;
@@ -72,15 +72,15 @@ void EIO_AfterOpen(uv_work_t* req) {
 
   v8::Handle<v8::Value> argv[2];
   if(data->errorString[0]) {
-    argv[0] = v8::Exception::Error(v8::String::New(data->errorString));
-    argv[1] = v8::Undefined();
+    argv[0] = v8::Exception::Error(NanNew<v8::String>(data->errorString));
+    argv[1] = NanUndefined();
     // not needed for AfterOpenSuccess
     delete data->dataCallback;
     delete data->errorCallback;
     delete data->disconnectedCallback;
   } else {
-    argv[0] = v8::Undefined();
-    argv[1] = v8::Int32::New(data->result);
+    argv[0] = NanUndefined();
+    argv[1] = NanNew<v8::Int32>(data->result);
     AfterOpenSuccess(data->result, data->dataCallback, data->disconnectedCallback, data->errorCallback);
   }
 
@@ -120,7 +120,7 @@ NAN_METHOD(Write) {
   WriteBaton* baton = new WriteBaton();
   memset(baton, 0, sizeof(WriteBaton));
   baton->fd = fd;
-  NanAssignPersistent(v8::Object, baton->buffer, buffer);
+  NanAssignPersistent<v8::Object>(baton->buffer, buffer);
   baton->bufferData = bufferData;
   baton->bufferLength = bufferLength;
   // baton->offset = 0;
@@ -153,11 +153,11 @@ void EIO_AfterWrite(uv_work_t* req) {
 
   v8::Handle<v8::Value> argv[2];
   if(data->errorString[0]) {
-    argv[0] = v8::Exception::Error(v8::String::New(data->errorString));
-    argv[1] = v8::Undefined();
+    argv[0] = v8::Exception::Error(NanNew<v8::String>(data->errorString));
+    argv[1] = NanUndefined();
   } else {
-    argv[0] = v8::Undefined();
-    argv[1] = v8::Int32::New(data->result);
+    argv[0] = NanUndefined();
+    argv[1] = NanNew<v8::Int32>(data->result);
   }
   data->callback->Call(2, argv);
 
@@ -179,7 +179,7 @@ void EIO_AfterWrite(uv_work_t* req) {
   }
   uv_mutex_unlock(&write_queue_mutex);
 
-  NanDispose(data->buffer);
+  NanDisposePersistent(data->buffer);
   delete data->callback;
   delete data;
   delete queuedWrite;
@@ -221,9 +221,9 @@ void EIO_AfterClose(uv_work_t* req) {
 
   v8::Handle<v8::Value> argv[1];
   if(data->errorString[0]) {
-    argv[0] = v8::Exception::Error(v8::String::New(data->errorString));
+    argv[0] = v8::Exception::Error(NanNew<v8::String>(data->errorString));
   } else {
-    argv[0] = v8::Undefined();
+    argv[0] = NanUndefined();
   }
   data->callback->Call(1, argv);
 
@@ -260,23 +260,23 @@ void EIO_AfterList(uv_work_t* req) {
 
   v8::Handle<v8::Value> argv[2];
   if(data->errorString[0]) {
-    argv[0] = v8::Exception::Error(v8::String::New(data->errorString));
-    argv[1] = v8::Undefined();
+    argv[0] = v8::Exception::Error(NanNew<v8::String>(data->errorString));
+    argv[1] = NanUndefined();
   } else {
-    v8::Local<v8::Array> results = v8::Array::New();
+    v8::Local<v8::Array> results = NanNew<v8::Array>();
     int i = 0;
     for(std::list<ListResultItem*>::iterator it = data->results.begin(); it != data->results.end(); ++it, i++) {
-      v8::Local<v8::Object> item = v8::Object::New();
-      item->Set(v8::String::New("comName"), v8::String::New((*it)->comName.c_str()));
-      item->Set(v8::String::New("manufacturer"), v8::String::New((*it)->manufacturer.c_str()));
-      item->Set(v8::String::New("serialNumber"), v8::String::New((*it)->serialNumber.c_str()));
-      item->Set(v8::String::New("pnpId"), v8::String::New((*it)->pnpId.c_str()));
-      item->Set(v8::String::New("locationId"), v8::String::New((*it)->locationId.c_str()));
-      item->Set(v8::String::New("vendorId"), v8::String::New((*it)->vendorId.c_str()));
-      item->Set(v8::String::New("productId"), v8::String::New((*it)->productId.c_str()));
+      v8::Local<v8::Object> item = NanNew<v8::Object>();
+      item->Set(NanNew<v8::String>("comName"), NanNew<v8::String>((*it)->comName.c_str()));
+      item->Set(NanNew<v8::String>("manufacturer"), NanNew<v8::String>((*it)->manufacturer.c_str()));
+      item->Set(NanNew<v8::String>("serialNumber"), NanNew<v8::String>((*it)->serialNumber.c_str()));
+      item->Set(NanNew<v8::String>("pnpId"), NanNew<v8::String>((*it)->pnpId.c_str()));
+      item->Set(NanNew<v8::String>("locationId"), NanNew<v8::String>((*it)->locationId.c_str()));
+      item->Set(NanNew<v8::String>("vendorId"), NanNew<v8::String>((*it)->vendorId.c_str()));
+      item->Set(NanNew<v8::String>("productId"), NanNew<v8::String>((*it)->productId.c_str()));
       results->Set(i, item);
     }
-    argv[0] = v8::Undefined();
+    argv[0] = NanUndefined();
     argv[1] = results;
   }
   data->callback->Call(2, argv);
@@ -326,11 +326,11 @@ void EIO_AfterFlush(uv_work_t* req) {
   v8::Handle<v8::Value> argv[2];
 
   if(data->errorString[0]) {
-    argv[0] = v8::Exception::Error(v8::String::New(data->errorString));
-    argv[1] = v8::Undefined();
+    argv[0] = v8::Exception::Error(NanNew<v8::String>(data->errorString));
+    argv[1] = NanUndefined();
   } else {
-    argv[0] = v8::Undefined();
-    argv[1] = v8::Int32::New(data->result);
+    argv[0] = NanUndefined();
+    argv[1] = NanNew<v8::Int32>(data->result);
   }
   data->callback->Call(2, argv);
 
@@ -376,11 +376,11 @@ void EIO_AfterDrain(uv_work_t* req) {
   v8::Handle<v8::Value> argv[2];
 
   if(data->errorString[0]) {
-    argv[0] = v8::Exception::Error(v8::String::New(data->errorString));
-    argv[1] = v8::Undefined();
+    argv[0] = v8::Exception::Error(NanNew<v8::String>(data->errorString));
+    argv[1] = NanUndefined();
   } else {
-    argv[0] = v8::Undefined();
-    argv[1] = v8::Int32::New(data->result);
+    argv[0] = NanUndefined();
+    argv[1] = NanNew<v8::Int32>(data->result);
   }
   data->callback->Call(2, argv);
 
@@ -392,23 +392,26 @@ void EIO_AfterDrain(uv_work_t* req) {
 SerialPortParity NAN_INLINE(ToParityEnum(const v8::Handle<v8::String>& v8str)) {
   NanScope();
 
-  v8::String::AsciiValue str(v8str);
-  if(!strcasecmp(*str, "none")) {
-    return SERIALPORT_PARITY_NONE;
+  size_t count;
+  char* str = NanCString(v8str, &count);
+
+  SerialPortParity parity = SERIALPORT_PARITY_NONE;
+
+  if(!strncasecmp(str, "none", count)) {
+    parity = SERIALPORT_PARITY_NONE;
+  } else if(!strncasecmp(str, "even", count)) {
+    parity = SERIALPORT_PARITY_EVEN;
+  } else if(!strncasecmp(str, "mark", count)) {
+    parity = SERIALPORT_PARITY_MARK;
+  } else if(!strncasecmp(str, "odd", count)) {
+    parity = SERIALPORT_PARITY_ODD;
+  } else if(!strncasecmp(str, "space", count)) {
+    parity = SERIALPORT_PARITY_SPACE;
   }
-  if(!strcasecmp(*str, "even")) {
-    return SERIALPORT_PARITY_EVEN;
-  }
-  if(!strcasecmp(*str, "mark")) {
-    return SERIALPORT_PARITY_MARK;
-  }
-  if(!strcasecmp(*str, "odd")) {
-    return SERIALPORT_PARITY_ODD;
-  }
-  if(!strcasecmp(*str, "space")) {
-    return SERIALPORT_PARITY_SPACE;
-  }
-  return SERIALPORT_PARITY_NONE;
+
+  delete[] str;
+
+  return parity;
 }
 
 SerialPortStopBits NAN_INLINE(ToStopBitEnum(double stopBits)) {

--- a/src/serialport_poller.cpp
+++ b/src/serialport_poller.cpp
@@ -35,24 +35,24 @@ void SerialportPoller::Init(Handle<Object> target) {
   NanScope();
 
   // Prepare constructor template
-  Local<FunctionTemplate> tpl = FunctionTemplate::New(New);
-  tpl->SetClassName(String::NewSymbol("SerialportPoller"));
+  Local<FunctionTemplate> tpl = NanNew<FunctionTemplate>(New);
+  tpl->SetClassName(NanNew<String>("SerialportPoller"));
   tpl->InstanceTemplate()->SetInternalFieldCount(1);
 
 
   // Prototype
 
   // SerialportPoller.close()
-  tpl->PrototypeTemplate()->Set(String::NewSymbol("close"),
-      FunctionTemplate::New(Close)->GetFunction());
+  tpl->PrototypeTemplate()->Set(NanNew<String>("close"),
+      NanNew<FunctionTemplate>(Close)->GetFunction());
 
   // SerialportPoller.start()
-  tpl->PrototypeTemplate()->Set(String::NewSymbol("start"),
-      FunctionTemplate::New(Start)->GetFunction());
+  tpl->PrototypeTemplate()->Set(NanNew<String>("start"),
+      NanNew<FunctionTemplate>(Start)->GetFunction());
 
-  NanAssignPersistent(v8::FunctionTemplate, serialportpoller_constructor, tpl);
+  NanAssignPersistent<FunctionTemplate>(serialportpoller_constructor, tpl);
 
-  target->Set(String::NewSymbol("SerialportPoller"), tpl->GetFunction());
+  target->Set(NanNew<String>("SerialportPoller"), tpl->GetFunction());
 }
 
 NAN_METHOD(SerialportPoller::New) {

